http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/ResourcesServiceInterp.scala

```diff
@@ -0,0 +1,358 @@
+package org.broadinstitute.dsde.workbench.leonardo.http.service
+
+import cats.Parallel
+import cats.effect.Async
+import cats.implicits.toFunctorOps
+import cats.mtl.Ask
+import cats.syntax.all._
+import org.broadinstitute.dsde.workbench.google2.DiskName
+import org.broadinstitute.dsde.workbench.leonardo.{
+  AppContext,
+  AppName,
+  AppStatus,
+  CloudContext,
+  DiskStatus,
+  RuntimeName,
+  WorkspaceId
+}
+import org.broadinstitute.dsde.workbench.leonardo.db.{
+  appQuery,
+  clusterQuery,
+  kubernetesClusterQuery,
+  nodepoolQuery,
+  persistentDiskQuery,
+  DbReference,
+  DiskServiceDbQueries,
+  KubernetesServiceDbQueries
+}
+import org.broadinstitute.dsde.workbench.leonardo.http.{
+  ctxConversion,
+  CreateAppRequest,
+  CreateDiskRequest,
+  CreateRuntimeRequest,
+  CreateRuntimeResponse,
+  GetAppResponse,
+  GetPersistentDiskResponse,
+  GetRuntimeResponse,
+  ListAppResponse,
+  ListPersistentDiskResponse,
+  ListRuntimeResponse2,
+  UpdateDiskRequest,
+  UpdateRuntimeRequest
+}
+import org.broadinstitute.dsde.workbench.leonardo.model.{
+  LeoAuthProvider,
+  NonDeletableDisksInProjectFoundException,
+  NonDeletableRuntimesInProjectFoundException
+}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.UserInfo
+
+import scala.concurrent.ExecutionContext
+
+final class ResourcesServiceInterp[F[_]: Parallel](authProvider: LeoAuthProvider[F])(implicit
+  F: Async[F],
+  dbRef: DbReference[F],
+  ec: ExecutionContext
+) extends ResourcesService[F]
+    with RuntimeService[F]
+    with AppService[F]
+    with DiskService[F] {
+  override def deleteAllResources(userInfo: UserInfo,
+                                  googleProject: GoogleProject,
+                                  deleteInCloud: Boolean,
+                                  deleteDisk: Boolean
+  )(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      _ <-
```

Comment:
Make sure you're checking appropriate permissions. I guess `deleteAllResourcesInCloud` will check each resource's permissions since we're reusing the existing `delete` code. But `deleteAllResourcesRecords` probably needs to do the same.